@here/harp-map-controls/lib/MapControls.ts

```diff
@@ -355,34 +355,34 @@ export class MapControls extends THREE.EventDispatcher {
             return;
         }
         this.camera.getWorldDirection(this.m_currentViewDirection);
-        const maxDistance = MapViewUtils.calculateDistanceToGroundFromZoomLevel(
+        let maxDistance = MapViewUtils.calculateDistanceToGroundFromZoomLevel(
             this.mapView,
             this.mapView.minZoomLevel
         );
-        const minDistance = MapViewUtils.calculateDistanceToGroundFromZoomLevel(
+        let minDistance = MapViewUtils.calculateDistanceToGroundFromZoomLevel(
             this.mapView,
             this.mapView.maxZoomLevel
         );
         this.m_currentViewDirection.multiplyScalar(amount);
         if (this.mapView.projection.type === geoUtils.ProjectionType.Planar) {
-            this.camera.position.z += this.m_currentViewDirection.z;
-            this.camera.position.z = Math.max(
+            const distance = THREE.Math.clamp(
+                this.camera.position.z + this.m_currentViewDirection.z,
                 minDistance,
-                Math.min(maxDistance, this.camera.position.z)
+                maxDistance
             );
+            this.camera.position.z = distance;
```

Comment:
I don't get what this method is doing. The name says it is moving the position along the view direction  but I don't see this happening(i.e. it only takes the z of the view direction).